### PR TITLE
contest & index  page test 

### PR DESCRIPTION
**Pull request by MadjarovaML**

### CODE REVIEW

File: automated_testing/test/constants/submissions-details-page-selectors.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const selectors = {
```
Comment (Minkov) [inline automated_testing/test/constants/submissions-details-page-selectors.ts:1]: Is this empty file neccessary?

**Review by Minkov — CHANGES_REQUESTED** [review]

File: automated_testing/test/constants/submissions-details-page-selectors.ts (hunk 1 above)
Comment (MadjarovaML) [reply automated_testing/test/constants/submissions-details-page-selectors.ts:1]: everything in submission-details was created in order to help Simona with the set up for her tests. 
I'm going to write in her branch and it'll not stay empty. If I have to, I will delete it here.

**Review by mishogv — DISMISSED** [review]

File: automated_testing/test/pageobjects/index-page.ts [hunk 2]
```diff
@@ -91,19 +87,19 @@ class IndexPage extends Page {
         return $(selectors.pageContestCardTimerSelector);
     }
 
-    public get competeCardButtonActivecontests() {
+    public get competeButtonInActiveContestCard() {
         return $(selectors.pageAllActiveCardsSelector).$$(selectors.pageCompeteButtonIdSelector);
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:91]: Won't this return all `compete buttons`? If so, the name or the indent should be changed, i.e.
- If you want a single button, change the selection
- If you want all buttons, change the name

File: automated_testing/test/pageobjects/index-page.ts [hunk 3]
```diff
@@ -91,19 +87,19 @@ class IndexPage extends Page {
...
     }
 
-    public get practiceCardButtoActiveContests() {
+    public get practiceButtonInActiveContestsCard() {
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:94]: Same as above.

File: automated_testing/test/pageobjects/index-page.ts [hunk 4]
```diff
@@ -91,19 +87,19 @@ class IndexPage extends Page {
...
         return $(selectors.pageAllActiveCardsSelector).$$(selectors.pagePracticeButtonIdSelector);
     }
 
-    public get competeCardButtonPastcontests() {
+    public get competeButtonInPastContestsCard() {
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:98]: Same as above

File: automated_testing/test/pageobjects/index-page.ts [hunk 5]
```diff
@@ -91,19 +87,19 @@ class IndexPage extends Page {
...
         return $(selectors.pageAllPastCardsSelector).$$(selectors.pageCompeteButtonIdSelector);
     }
 
-    public get practiceCardButtoPastContests() {
+    public get practiceButtoInPastContestsCard() {
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:102]: Same as above.

File: automated_testing/test/pageobjects/index-page.ts [hunk 6]
```diff
@@ -112,7 +108,7 @@ class IndexPage extends Page {
     }
 
     public get firstCompeteCardButtonActiveContests() {
-        return $(selectors.pageAllActiveCardsSelector).$(selectors.pageCompeteButtonIdSelector);
+        return $(selectors.pageAllActiveCardsSelector).$$(selectors.pageCompeteButtonIdSelector)[0];
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:111]: Isn't this the same logic as in `competeButtonInActiveContestCard()`?
Maybe just do `competeButtonInActiveContestCard()[0]`

File: automated_testing/test/pageobjects/index-page.ts [hunk 7]
```diff
@@ -123,6 +119,22 @@ class IndexPage extends Page {
         return $(selectors.pageSeePastContestsButtonSelector);
     }
 
+    public get secondCompeteCardButtonActiveContests() {
+        return $(selectors.pageAllActiveCardsSelector).$$(selectors.pageCompeteButtonIdSelector)[1];
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:123]: Isn't this the same logic as in `competeButtonInActiveContestCard()`?
Maybe just do `competeButtonInActiveContestCard()[1]`

File: automated_testing/test/pageobjects/index-page.ts [hunk 8]
```diff
@@ -123,6 +119,22 @@ class IndexPage extends Page {
...
+    }
+
+    public get secondPracticeCardButtoPastContests() {
+        return $(selectors.pageAllPastCardsSelector).$$(selectors.pagePracticeButtonIdSelector)[1];
```
Comment (Minkov) [inline automated_testing/test/pageobjects/index-page.ts:127]: Isn't this the same logic as in `practiceButtoInPastContestsCard()`?
Maybe just do `practiceButtoInPastContestsCard()[0]``?

**Review by Minkov — CHANGES_REQUESTED** [review]

File: automated_testing/test/pageobjects/index-page.ts (hunk 2 above)
Comment (MadjarovaML) [reply automated_testing/test/pageobjects/index-page.ts:91]: I will change the name

File: automated_testing/test/pageobjects/index-page.ts (hunk 8 above)
Comment (MadjarovaML) [reply automated_testing/test/pageobjects/index-page.ts:127]: It is the second practice button. That is why I used `$$(selectors.pagePracticeButtonIdSelector)[1]; `
The same logic goes for the second compete button. I can delete those selectors as they were used when I didn't have a local DB set up.

File: automated_testing/test/pageobjects/submissions-details-page.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable class-methods-use-this */
```
Comment (Minkov) [inline automated_testing/test/pageobjects/submissions-details-page.ts:1]: No.

**Review by Minkov — CHANGES_REQUESTED** [review]

**Review by Minkov — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

**Pull request merged**